api/sip/client.py

```diff
@@ -874,3 +884,20 @@ def read_message(self, max_size=1024*1024):
 
     def disconnect(self):
         pass
+
+
+class MockSIPClientFactory(object):
+    """Pass this into SIP2AuthenticationProvider to instantiate a
+    MockSIPClient based on its configuration, _and_ to use that
+    MockSIPClient every single time; normally
+    SIP2AuthenticationProvider will instantiate a different client for
+    every simulated server interaction, making it impossible to queue
+    responses or look at the results.
+    """
+    def __init__(self):
+        self.client = None
+
+    def __call__(self, **kwargs):
+        if self.client is None:
+            self.client = MockSIPClient(**kwargs)
+        return self.client
```

Comment:
What we had before worked, but it's one of those situations where I'm not really sure how. This makes it really easy to write a test where you set up an `ExternalIntegration` with a certain configuration and then pass it into a `SIP2AuthenticationProvider`. With `MockSIPClientFactory` you can then run tests on a mocked-up `SIPClient` that is a) persistent across requests (something we wouldn't want for a real client) and b) is instantiated with the `ExternalIntegration` settings you configured at the start of the test.